sql/api/src/main/scala/org/apache/spark/sql/types/StringType.scala

```diff
@@ -30,7 +30,7 @@ import org.apache.spark.sql.catalyst.util.CollationFactory
  *   The id of collation for this StringType.
  */
 @Stable
-class StringType private (val collationId: Int) extends AtomicType with Serializable {
+class StringType private[sql] (val collationId: Int) extends AtomicType with Serializable {
```

Comment:
no need to do this, as we have `def apply(collationId: Int)` in `object StringType`